olp-cpp-sdk-core/src/http/adapters/HarCaptureAdapter.cpp

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (C) 2025 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/core/http/adapters/HarCaptureAdapter.h>
+
+#include <rapidjson/ostreamwrapper.h>
+#include <rapidjson/prettywriter.h>
+
+#include <deque>
+#include <fstream>
+#include <iomanip>
+#include <mutex>
+#include <sstream>
+#include <unordered_map>
+#include <utility>
+
+#include "olp/core/logging/Log.h"
+
+namespace olp {
+namespace http {
+namespace {
+constexpr std::hash<std::string> hasher{};
```

Comment:
Does it make sense? Hasher is stateless, and I believe it should be inlined anyway